cloud_governance/common/aws/cost_explorer/cost_explorer_operations.py

```diff
@@ -12,21 +12,22 @@ class CostExplorerOperations:
     def __init__(self):
         self.cost_explorer_client = boto3.client('ce')
 
-    def get_daily_cost_usage(self, tag: str, metrics_type: str):
+    def get_daily_cost_usage(self, tag: str, report_type: str, start_date: str = '', end_date: str = ''):
         """
         This method extracts the price by Tag provided
         @return:
         """
-        today = datetime.now()
-        start_time = today - timedelta(1)
-        today = str(today.strftime('%Y-%m-%d'))
-        start_time = str(start_time.strftime('%Y-%m-%d'))
+        if not start_date and not end_date:
+            end_date = datetime.now()
+            start_date = end_date - timedelta(1)
+            start_date = str(start_date.strftime('%Y-%m-%d'))
+            end_date = str(end_date.strftime('%Y-%m-%d'))
         return self.cost_explorer_client.get_cost_and_usage(
             TimePeriod={
-                'Start': start_time,
-                'End': today
+                'Start': start_date,
+                'End': end_date
             },
-            Granularity='DAILY',
-            Metrics=['BlendedCost'],
+            Granularity=report_type,
```

Comment:
1. granularity instead of report_type
2. cost_metric parameter instead of hard coded 'BlendedCost'

You should get granularity and cost_metric by parameters from environment variable